src/v/redpanda/admin/server.cc

```diff
@@ -2106,6 +2108,31 @@ admin_server::update_user_handler(std::unique_ptr<ss::http::request> req) {
     co_return ss::json::json_return_type(ss::json::json_void());
 }
 
+ss::future<ss::json::json_return_type>
+admin_server::oidc_whoami_handler(std::unique_ptr<ss::http::request> req) {
+    auto auth_hdr = req->get_header("authorization");
+    if (!auth_hdr.starts_with("Bearer ")) {
+        throw ss::httpd::base_exception{
+          "Invalid Authorization header",
+          ss::http::reply::status_type::unauthorized};
+    }
+
+    security::oidc::authenticator auth{_controller->get_oidc_service().local()};
+    auto res = auth.authenticate(auth_hdr.substr(7));
```

Comment:
>auth_hdr.substr(7)

nit: probably overkill to add any real abstraction, but the number `7` here contains arcane knowledge. more an observation than a nit I guess. (it's just len("Bearer") right?)

Reply:
> > auth_hdr.substr(7)
> 
> nit: probably overkill to add any real abstraction, but the number `7` here contains arcane knowledge. more an observation than a nit I guess. (it's just len("Bearer") right?)

Yeah, I've been wondering about this a bit, but it seemed overkill. I'll add the abstraction.

Reply:
>> the number 7 here contains arcane knowledge
>
> I don't think it's really worth writing the abstraction.

Added some constants to request_auth and used them.